tests/oeedger8r/host/testmisc.cpp

```diff
@@ -17,11 +17,19 @@ void run_misc_tests(oe_enclave_t* enclave)
 
     int8_t arr[25];
 
+    // In calls below, the enclave memory pointer is passed in
+    // as input array. With previous marshaling scheme, the pointer
+    // is passed all the way to enclave which returns OE_INVALID_PARAMETER.
+    // With the new marshaling scheme, the array is copied into a flat buffer
+    // in the host itself, and the buffer is passed in to the enclave and
+    // the call works fine. Note: It is OK to read enclave memory; the value
+    // read will be a fixed pattern/constant.
```

Comment:
Can you please explain this comment since I don't understand the security implication you are making.

Reply:
I would argue that it would be cleaner to throw an error on the host side, since the behavior might not be the same across all enclave architectures and it is non-intuitive even on SGX. But this is the call of the team in the end of the day.

Reply:
I would argue that this behavior is highly SGX specific and even there it is not intuitive (after a year of not working with the technology I remembered that it threw an access violation). The host could easily check the pointer and throw a user-friendly error. But in the end of the day, this is the call of the team.

Reply:
I would argue that this behavior is highly SGX specific and even there it is not intuitive (after a year of not working with the technology I remembered that it threw an access violation). The host could easily check the pointer and throw a user-friendly error. But in the end of the day, this is the call of the team.

Reply:
I am recommending that the old behavior in the unit test is maintained. Passing an invalid pointer to the ECALL should result in the error code OE_INVALID_PARAMETER.

With the new changes in place that would mean that the host would perform the check to avoid attempting to read memory it should not have access to. This is not about security, as you clarified above, but it would still lead to a more consistent and portable behavior.

I just don't consider it a reasonable requirement for an enclave environment to allow read to enclave memory with just the value being replace by a constant. This particular behavior is very specific to Intel SGX (which I even forget happened after 1 year of not working on the platform) and with the current change the unit test will have a strong dependence on this behavior.

Reply:
Alternatively, just remove this particular test altogether since it serves no purpose after this change-set is applied and the behavior for invalid pointers can be explored later.

Reply:
Thanks for the clarification. As suggested, I am removing the test - it causes confusion and is no longer applicable.